openedx/custom/db/fields.py

```diff
@@ -9,6 +9,11 @@
 from openedx.custom.helpers import add_other_field_in_choices
 
 
+class OtherMultiSelectFieldList(MSFList):
+    def __str__(self):
+        selected_choice_list = [self.choices.get(int(i)) if i.isdigit() else (self.choices.get(i) or i) for i in self]
```

Comment:
Why don't we  use
```
self.choices.get(i) for i in self
```

Reply:
if the key is a digit, in that case, it will be integer that why here we need to cast, other then that. we have to handle `(self.choices.get(i) or i)`